File: invesdwin-scripting-parent/invesdwin-scripting-julia/invesdwin-scripting-julia-runtime-contract/src/test/java/de/invesdwin/scripting/julia/runtime/contract/InputsAndResultsTestString.jl
println("getString")
if isdefined(Main, :getString) && !isnothing(getString)
	error("getString already defined!")
end
getString = putString
println(typeof(getString))
println(getString)
if typeof(getString) != String
	error("getString not String!")
end

println("getStringWithNull")
if isdefined(Main, :getStringWithNull) && !isnothing(getStringWithNull)
	error("getStringWithNull already defined!")
end
getStringWithNull = putStringWithNull
println(typeof(getStringWithNull))
println(getStringWithNull)
if typeof(getStringWithNull) != Nothing
	error("getStringWithNull not Nothing!")
end
if !isnothing(getStringWithNull)
	error("getStringWithNull not nothing!")
end

println("getStringVector")
if isdefined(Main, :getStringVector) && !isnothing(getStringVector)
	error("getStringVector already defined!")
end
getStringVector = putStringVector
println(eltype(getStringVector))
println(getStringVector)
if eltype(getStringVector) != String
	error("getStringVector not String!")
end


println("getStringVectorWithNull")
if isdefined(Main, :getStringVectorWithNull) && !isnothing(getStringVectorWithNull)
	error("getStringVectorWithNull already defined!")
end
getStringVectorWithNull = putStringVectorWithNull
println(eltype(getStringVectorWithNull))
println(getStringVectorWithNull)
if eltype(getStringVectorWithNull) != String
	error("getStringVectorWithNull not String!")
end
if !isempty(getStringVectorWithNull[2])
	error("getStringVectorWithNull[2] not empty!")
end

println("getStringVectorAsList")
if isdefined(Main, :getStringVectorAsList) && !isnothing(getStringVectorAsList)
	error("getStringVectorAsList already defined!")
end
getStringVectorAsList = putStringVectorAsList
println(eltype(getStringVectorAsList))
println(getStringVectorAsList)
if eltype(getStringVectorAsList) != String
	error("getStringVectorAsList not String!")
end

println("getStringVectorAsListWithNull")
if isdefined(Main, :getStringVectorAsListWithNull) && !isnothing(getStringVectorAsListWithNull)
	error("getStringVectorAsListWithNull already defined!")
end
getStringVectorAsListWithNull = putStringVectorAsListWithNull
println(eltype(getStringVectorAsListWithNull))
println(getStringVectorAsListWithNull)
if eltype(getStringVectorAsListWithNull) != String
	error("getStringVectorAsListWithNull not String!")
end
if !isempty(getStringVectorAsListWithNull[2])
	error("getStringVectorAsListWithNull[2] not empty!")
end

println("getStringMatrix")
if isdefined(Main, :getStringMatrix) && !isnothing(getStringMatrix)
	error("getStringMatrix already defined!")
end
getStringMatrix = putStringMatrix
println(eltype(getStringMatrix))
println(getStringMatrix)
if eltype(getStringMatrix) != String
	error("getStringMatrix not String!")
end


println("getStringMatrixWithNull")
if isdefined(Main, :getStringMatrixWithNull) && !isnothing(getStringMatrixWithNull)
	error("getStringMatrixWithNull already defined!")
end
getStringMatrixWithNull = putStringMatrixWithNull
println(eltype(getStringMatrixWithNull))
println(getStringMatrixWithNull)
if eltype(getStringMatrixWithNull) != String
	error("getStringMatrixWithNull not String!")
end
if !isempty(getStringMatrixWithNull[1,1])
	error("getStringMatrixWithNull[1,1] not empty!")
end
if !isempty(getStringMatrixWithNull[2,2])
	error("getStringMatrixWithNull[2,2] not empty!")
end
if !isempty(getStringMatrixWithNull[3,3])
	error("getStringMatrixWithNull[3,3] not empty!")
end

println("getStringMatrixAsList")
if isdefined(Main, :getStringMatrixAsList) && !isnothing(getStringMatrixAsList)
	error("getStringMatrixAsList already defined!")
end
getStringMatrixAsList = putStringMatrixAsList
println(eltype(getStringMatrixAsList))
println(getStringMatrixAsList)
if eltype(getStringMatrixAsList) != String
	error("getStringMatrixAsList not String!")
end

println("getStringMatrixAsListWithNull")
if isdefined(Main, :getStringMatrixAsListWithNull) && !isnothing(getStringMatrixAsListWithNull)
	error("getStringMatrixAsListWithNull already defined!")
end
getStringMatrixAsListWithNull = putStringMatrixAsListWithNull
println(eltype(getStringMatrixAsListWithNull))
println(getStringMatrixAsListWithNull)
if eltype(getStringMatrixAsListWithNull) != String
	error("getStringMatrixAsListWithNull not String!")
end
if !isempty(getStringMatrixAsListWithNull[1,1])
	error("getStringMatrixAsListWithNull[1,1] not empty!")
end
if !isempty(getStringMatrixAsListWithNull[2,2])
	error("getStringMatrixAsListWithNull[2,2] not empty!")
end
if !isempty(getStringMatrixAsListWithNull[3,3])
	error("getStringMatrixAsListWithNull[3,3] not empty!")
end

File: invesdwin-scripting-parent/invesdwin-scripting-runtime-lua/src/test/java/de/invesdwin/scripting/runtime/lua/tests/InputsAndResultsTestString.lua
arraysToString = java.method(java.import('de.invesdwin.util.collections.Arrays'),'toString','java.lang.Object')

print("getString")
if(getString ~= nil) then
	error("getString already defined!")
end
getString = putString
getStringType = type(getString)
print(getStringType)
print(getString)
if(getStringType ~= "string") then
	error("getString not String!")
end

print("getStringWithNull")
if(getStringWithNull ~= nil) then
	error("getStringWithNull already defined!")
end
getStringWithNull = putStringWithNull
getStringWithNullType = type(getStringWithNull)
print(getStringWithNullType)
print(getStringWithNull)
if(getStringWithNull ~= nil) then
	error("getStringWithNull not null!")
end

print("getStringVector")
if(getStringVector ~= nil) then
	error("getStringVector already defined!")
end
getStringVector = putStringVector
getStringVectorType = type(getStringVector[1])
print(getStringVectorType)
print(arraysToString(getStringVector))
if(getStringVectorType ~= "string") then
	error("getStringVector not String!")
end

print("getStringVectorWithNull")
if(getStringVectorWithNull ~= nil) then
	error("getStringVectorWithNull already defined!")
end
getStringVectorWithNull = putStringVectorWithNull
getStringVectorWithNullType = type(getStringVectorWithNull[1])
print(getStringVectorWithNullType)
print(arraysToString(getStringVectorWithNull))
if(getStringVectorWithNullType ~= "string") then
	error("getStringVectorWithNull not String!")
end
if(getStringVectorWithNull[2] ~= nil) then
	error("getStringVectorWithNull[1] not null!")
end

print("getStringVectorAsList")
if(getStringVectorAsList ~= nil) then
	error("getStringVectorAsList already defined!")
end
getStringVectorAsList = putStringVectorAsList
getStringVectorAsListType = type(getStringVectorAsList[1])
print(getStringVectorAsListType)
print(arraysToString(getStringVectorAsList))
if(getStringVectorAsListType ~= "string") then
	error("getStringVectorAsList not String!")
end

print("getStringVectorAsListWithNull")
if(getStringVectorAsListWithNull ~= nil) then
	error("getStringVectorAsListWithNull already defined!")
end
getStringVectorAsListWithNull = putStringVectorAsListWithNull
getStringVectorAsListWithNullType = type(getStringVectorAsListWithNull[1])
print(getStringVectorAsListWithNullType)
print(arraysToString(getStringVectorAsListWithNull))
if(getStringVectorAsListWithNullType ~= "string") then
	error("getStringVectorAsListWithNull not String!")
end
if(getStringVectorAsListWithNull[2] ~= nil) then
	error("getStringVectorAsListWithNull[1] not null!")
end

print("getStringMatrix")
if(getStringMatrix ~= nil) then
	error("getStringMatrix already defined!")
end
getStringMatrix = putStringMatrix
getStringMatrixType = type(getStringMatrix[1][1])
print(getStringMatrixType)
print(arraysToString(getStringMatrix))
if(getStringMatrixType ~= "string") then
	error("getStringMatrix not String!")
end

print("getStringMatrixWithNull")
if(getStringMatrixWithNull ~= nil) then
	error("getStringMatrixWithNull already defined!")
end
getStringMatrixWithNull = putStringMatrixWithNull
getStringMatrixWithNullType = type(getStringMatrixWithNull[1][2])
print(getStringMatrixWithNullType)
print(arraysToString(getStringMatrixWithNull))
if(getStringMatrixWithNullType ~= "string") then
	error("getStringMatrixWithNull not String!")
end
if(getStringMatrixWithNull[1][1] ~= nil) then
	error("getStringMatrixWithNull[0][0] not null!")
end
if(getStringMatrixWithNull[2][2] ~= nil) then
	error("getStringMatrixWithNull[1][1] not null!")
end
if(getStringMatrixWithNull[3][3] ~= nil) then
	error("getStringMatrixWithNull[2][2] not null!")
end

print("getStringMatrixAsList")
if(getStringMatrixAsList ~= nil) then
	error("getStringMatrixAsList already defined!")
end
getStringMatrixAsList = putStringMatrixAsList
getStringMatrixAsListType = type(getStringMatrixAsList[1][1])
print(getStringMatrixAsListType)
print(arraysToString(getStringMatrixAsList))
if(getStringMatrixAsListType ~= "string") then
	error("getStringMatrixAsList not String!")
end

print("getStringMatrixAsListWithNull")
if(getStringMatrixAsListWithNull ~= nil) then
	error("getStringMatrixAsListWithNull already defined!")
end
getStringMatrixAsListWithNull = putStringMatrixAsListWithNull
getStringMatrixAsListWithNullType = type(getStringMatrixAsListWithNull[1][2])
print(getStringMatrixAsListWithNullType)
print(arraysToString(getStringMatrixAsListWithNull))
if(getStringMatrixAsListWithNullType ~= "string") then
	error("getStringMatrixAsListWithNull not String!")
end
if(getStringMatrixAsListWithNull[1][1] ~= nil) then
	error("getStringMatrixAsListWithNull[0][0] not null!")
end
if(getStringMatrixAsListWithNull[2][2] ~= nil) then
	error("getStringMatrixAsListWithNull[1][1] not null!")
end
if(getStringMatrixAsListWithNull[3][3] ~= nil) then
	error("getStringMatrixAsListWithNull[2][2] not null!")
end
